### PR TITLE
[develop] Add restart capability to forecast task

### LINKED ISSUE #681: Add restart capability to forecast task
## Description
- The ufs weather model has the restart (warm-start) capability, but this option is not available in the UFS SRW App currently.
- This ability is the NCO's requirement for the operational implementation.

## Solution
Add the restart option to the `run_fcst` task to use the warm start option of the ufs weather model.

## Acceptance Criteria (Definition of Done)
The forecast task can start again from the specific forecast hour that the restart files exist for in the 'RESTART' directory.

### PR DESCRIPTION
**Pull request by chan-hoo**
## DESCRIPTION OF CHANGES: 
- Add a flag `DO_FCST_RESTART` turning on/off the restart option of the forecast task.
- Add a python script `update_restart_input_nml_file.py` replacing the six parameters related to the restart option in the FV3 input.nml file.
- Add a parameter `fhrot` to model_configure.
- Update the configuration files `model_configure` and `nems.configure` with the latest format.
- This capability is the NCO's requirement.

### Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## TESTS CONDUCTED: 

- In the following example, the restart run starts from fhr=003.
- The sample run uses 'FV3_GFS_v15_thompson_mynn_lam3km' to match with the regression test of the ufs weather model.
1. Build the app without `-a` argument:
```
./devbuild.sh -p=hera
```

2. Run the following configuration file:
```
metadata:
  description: This script is to test the restart capability.
user:
  RUN_ENVIR: community
  MACHINE: HERA
  ACCOUNT: naqfc
platform:
  BUILD_MOD_FN: build_hera_intel
workflow:
  USE_CRON_TO_RELAUNCH: true
  CRON_RELAUNCH_INTVL_MNTS: 3
  EXPT_SUBDIR: test_restart_lam3km
  CCPP_PHYS_SUITE: FV3_GFS_v15_thompson_mynn_lam3km
  PREDEF_GRID_NAME: RRFS_CONUS_3km
  DATE_FIRST_CYCL: '2022070100'
  DATE_LAST_CYCL: '2022070100'
  FCST_LEN_HRS: 6
  PREEXISTING_DIR_METHOD: rename
  VERBOSE: false
  DEBUG: false
  COMPILER: intel
task_get_extrn_ics:
  EXTRN_MDL_NAME_ICS: FV3GFS
  FV3GFS_FILE_FMT_ICS: netcdf
task_get_extrn_lbcs:
  EXTRN_MDL_NAME_LBCS: FV3GFS
  FV3GFS_FILE_FMT_LBCS: netcdf
  LBC_SPEC_INTVL_HRS: 3
task_run_fcst:
  RESTART_INTERVAL: 3
  DO_FCST_RESTART: true
```

3. Run the `rocotorewind` for the forecast task:
```
rocotorewind -w FV3LAM_wflow.xml -d FV3LAM_wflow.db -v 10 -c 202207010000 -t run_fcst_mem000
``` 

4. Move the result files to another directory:
```
cd 2022070100
mkdir RES_ORG
mv dynf* RES_ORG
mv phyf* RES_ORG
``` 

5. Re-run the forecast task:
```
./launch_FV3LAM_wflow.sh
``` 

6. Check that `phyf004-006.nc` and `dynf004-006.nc` files are produced again.

- [x] hera.intel
- [ ] orion.intel
- [ ] cheyenne.intel
- [ ] cheyenne.gnu
- [ ] gaea.intel
- [ ] jet.intel
- [x] wcoss2.intel
- [ ] NOAA Cloud (indicate which platform)
- [ ] Jenkins
- [ ] fundamental test suite
- [ ] comprehensive tests (specify *which* if a subset was used)

## ISSUE: 
Fixes issue mentioned in #681 

## DEPENDENCIES:
PR https://github.com/ufs-community/ufs-srweather-app/pull/663

## CHECKLIST

- [x] My code follows the style guidelines in the Contributor's Guide
- [x] I have performed a self-review of my own code using the Code Reviewer's Guide
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes need updates to the documentation. I have made corresponding changes to the documentation
- [ ] My changes do not require updates to the documentation (explain).
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [ ] Any dependent changes have been merged and published

## CONTRIBUTORS: 
RatkoVasic-NOAA

### CODE REVIEW

**Comment by danielabdi-noaa** [comment]
@chan-hoo Looks good to me. I am running the test config you provided, will let you know how it goes!

**Comment by chan-hoo** [comment]
Thank you, @danielabdi-noaa !

**Comment by chan-hoo** [comment]
@danielabdi-noaa, @MichaelLueken, you can find the reproducibility of the restart run here: https://docs.google.com/presentation/d/1ru7t_8EI8i0fiSCdnV6t0xwvMkBHBuo-1Z3vsnE_kqg/edit?usp=sharing

**Review by danielabdi-noaa — APPROVED** [review]
@chan-hoo Unusually long queue today on hera so couldn't re-run fcst. Thanks for the slides!
RRFS_dev also has its own way of handling re-starts so we would need to coordinate to have one way of handling restarts. Approving now!

**Comment by danielabdi-noaa** [comment]
@chan-hoo Forecast is failing for me for some reason. I've checked out and rebuilt all the binaries but that didn't help.
Here is the directory of my test on Hera:
```
/scratch2/BMC/gsd-hpcs/Daniel.Abdi/expt_dirs/test_restart_lam3km
```

File: devbuild.sh
Comment (christinaholtNOAA) [inline devbuild.sh]: We likely don't want to remove these options. Is this feature broken for more recent versions of the model?

File: parm/FV3.input.yml
Comment (christinaholtNOAA) [inline parm/FV3.input.yml]: These changes will change the answer for all configurations. We need to make sure that that is welcome, and that all configurations using this package will pass the test.

File: parm/model_configure
Comment (christinaholtNOAA) [inline parm/model_configure:23]: Are these changes appropriate for all configurations of SRW?

File: scripts/exregional_make_orog.sh
Comment (christinaholtNOAA) [inline scripts/exregional_make_orog.sh]: This syntax might be easier with:

```
suites=( FV3_HRRR FV3_GFS_v15_thompson_mynn_lam3km  FV3_GFS_v17_p8 )
if [[ ${suites[@]} =~ "${CCPP_PHYS_SUITE}" ]] ; then
   ....
```

File: scripts/exregional_run_fcst.sh
Comment (christinaholtNOAA) [inline scripts/exregional_run_fcst.sh]: `org` is unclear here. How about `orig`?

File: scripts/exregional_run_fcst.sh
Comment (christinaholtNOAA) [inline scripts/exregional_run_fcst.sh]: `fileids` is a bit tough to read. Please consider separating the words with an underscore`file_ids`.

File: sorc/CMakeLists.txt
Comment (christinaholtNOAA) [inline sorc/CMakeLists.txt]: Is this supposed to be on for all configurations? Is it on by default already in the model?

**Review by christinaholtNOAA — CHANGES_REQUESTED** [review]

File: devbuild.sh
Comment (chan-hoo) [reply devbuild.sh]: @christinaholtNOAA, Have you tried these options? These two don't work with the recent version of the ufs weather model for me. Currently, we set these two options in `sorc/CMakeLists.txt`.

File: parm/FV3.input.yml
Comment (chan-hoo) [reply parm/FV3.input.yml]: This is the recommendation of the physics group at EMC.

File: parm/model_configure
Comment (chan-hoo) [reply parm/model_configure:23]: It should work with the hash of the ufs weather model in SRW. All the regional regression tests in the ufs weather model have these parameters in 'model_configure'. Which configuration do you think will have an issue with them?

File: scripts/exregional_make_orog.sh
Comment (chan-hoo) [reply scripts/exregional_make_orog.sh]: Updated.

File: scripts/exregional_run_fcst.sh
Comment (chan-hoo) [reply scripts/exregional_run_fcst.sh]: Updated.

File: scripts/exregional_run_fcst.sh
Comment (chan-hoo) [reply scripts/exregional_run_fcst.sh]: Updated.

File: sorc/CMakeLists.txt
Comment (chan-hoo) [reply sorc/CMakeLists.txt]: Removed.

File: parm/FV3.input.yml
Comment (chan-hoo) [reply parm/FV3.input.yml]: @yangfanglin, could you explain why this value is better than the previous one (1.0e-05)? @christinaholtNOAA is worried about this change.

**Comment by chan-hoo** [comment]
@danielabdi-noaa, did you use the coupled ufs weather model (compiled with -a=ATMAQ)? The above sample script is for the ATM standalone model. You should build it without the `-a` argument. I am sorry if my branch name `aqm_restart` made you confused. This PR is not only for AQM but also for stand-alone ATM model (FV3-LAM).

**Comment by danielabdi-noaa** [comment]
@danielabdi-noaa Yep, I did build for AQM. Will try again after building for standard SRW.

File: parm/FV3.input.yml
Comment (yangfanglin) [reply parm/FV3.input.yml]: @chan-hoo  Correction:   In your table you shared with me the two values are actually the same,  rthresh = 1e-05 in SRWapp and rthresh = 10.e-6 in global workflow.    The default value in model code is 10.e-6

File: devbuild.sh
Comment (chan-hoo) [reply devbuild.sh]: change removed.

File: parm/FV3.input.yml
Comment (chan-hoo) [reply parm/FV3.input.yml]: Removed.

**Comment by danielabdi-noaa** [comment]
@chan-hoo No luck, forecast failed again. The result is in the same place.

**Comment by chan-hoo** [comment]
@danielabdi-noaa, uhm... strange... the input namelist files look the same each other:
```
(regional_workflow) [Chan-hoo.Jeon@hfe03 2022070100]$ diff /scratch2/BMC/gsd-hpcs/Daniel.Abdi/expt_dirs/test_restart_lam3km/2022070100/input.nml input.nml_orig
263,267c263,267
<     fnaisc = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/CFSR.SEAICE.1982.2012.monthly.clim.grb'
<     fnglac = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/global_glacier.2x2.grb'
<     fnmskh = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/seaice_newland.grb'
<     fnmxic = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/global_maxice.2x2.grb'
<     fnsmcc = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/global_soilmgldas.t126.384.190.grb'
---
>     fnaisc = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/CFSR.SEAICE.1982.2012.monthly.clim.grb'
>     fnglac = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/global_glacier.2x2.grb'
>     fnmskh = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/seaice_newland.grb'
>     fnmxic = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/global_maxice.2x2.grb'
>     fnsmcc = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/global_soilmgldas.t126.384.190.grb'
269c269
<     fnsnoc = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/global_snoclim.1.875.grb'
---
>     fnsnoc = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/global_snoclim.1.875.grb'
271c271
<     fntsfc = '../../../../../det/UFS_SRW_App/develop/fix/fix_am/RTGSST.1982.2012.monthly.clim.grb'
---
>     fntsfc = '../../../../../../../../BMC/det/UFS_SRW_App/develop/fix/fix_am/RTGSST.1982.2012.monthly.clim.grb'
(regional_workflow) [Chan-hoo.Jeon@hfe03 2022070100]$ diff /scratch2/BMC/gsd-hpcs/Daniel.Abdi/expt_dirs/test_restart_lam3km/2022070100/model_configure model_configure_orig
(regional_workflow) [Chan-hoo.Jeon@hfe03 2022070100]$ diff /scratch2/BMC/gsd-hpcs/Daniel.Abdi/expt_dirs/test_restart_lam3km/2022070100/nems.configure nems.configure       
```
I ran the task with your `ufs_model` but it failed too.
I guess something was wrong when you compiled it.
Can you copy my `ufs_model` and run the forecast task again?
```
/scratch2/NCEPDEV/fv3-cam/Chan-hoo.Jeon/ufs_srw_app/srw_dev_test/ufs-srweather-app/exec
```

**Comment by danielabdi-noaa** [comment]
@chan-hoo Yes it seems to be working with your binary! I built the binaries with the following command
```
./devbuild.sh -p=hera all --remove
```
and made sure the ufs-weather-model hash is what is secified in `External.cfg` which is `b388eb9`.
I will re-build and try again but I think my origanal build used the same hash ...

**Comment by danielabdi-noaa** [comment]
@chan-hoo Nevermind, I removed my ufs clone and re-built again and it works now. Sorry about the hassle.

File: parm/model_configure
Comment (christinaholtNOAA) [reply parm/model_configure:23]: I have no idea. I don't know what they do, I just want to make sure that they are generic enough not to cause problems with other configurations.

**Review by christinaholtNOAA — APPROVED** [review]
Thanks for making the requested changes!

File: parm/model_configure
Comment (chan-hoo) [reply parm/model_configure:23]: I made this change based on the 'model_configure' files of the regression tests in the corresponding hash of the ufs weather model. I think they won't make any problems with any configurations.

**Comment by chan-hoo** [comment]
@danielabdi-noaa, @christinaholtNOAA, thank you for your approvals!

**Review by MichaelLueken — APPROVED** [review]
@chan-hoo These changes look good to me!

**Comment by MichaelLueken** [comment]
@chan-hoo Manually running the WE2E fundamental tests on Jet (since Jenkins is currently unable to communicate with the machine), the `grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2` test failed in the `run_fcst` task with the following message in the log:

```
NOTE from PE     0: reading topographic/orographic information from INPUT/oro_data.tile*.nc
NOTE from PE     0: reading surface properties data from INPUT/sfc_data.tile*.nc

FATAL from PE     0: NetCDF: Variable not found: get_variable_num_dimension: file:INPUT/sfc_data.nc variable: tref

Image              PC                Routine            Line        Source
ufs_model          0000000005DF19C7  Unknown               Unknown  Unknown
ufs_model          000000000354C5F9  mpp_mod_mp_mpp_er          72  mpp_util_mpi.inc
ufs_model          000000000342C5D1  fms_io_utils_mod_         190  fms_io_utils.F90
ufs_model          000000000347DAB5  netcdf_io_mod_mp_         368  netcdf_io.F90
```

The rest of the tests passed on Jet.  Do you know what might be causing this issue while testing this PR on Jet?

**Comment by JeffBeck-NOAA** [comment]
> @chan-hoo Manually running the WE2E fundamental tests on Jet (since Jenkins is currently unable to communicate with the machine), the `grid_RRFS_CONUS_25km_ics_GSMGFS_lbcs_GSMGFS_suite_GFS_v15p2` test failed in the `run_fcst` task with the following message in the log:
> 
> ```
> NOTE from PE     0: reading topographic/orographic information from INPUT/oro_data.tile*.nc
> NOTE from PE     0: reading surface properties data from INPUT/sfc_data.tile*.nc
> 
> FATAL from PE     0: NetCDF: Variable not found: get_variable_num_dimension: file:INPUT/sfc_data.nc variable: tref
> 
> Image              PC                Routine            Line        Source
> ufs_model          0000000005DF19C7  Unknown               Unknown  Unknown
> ufs_model          000000000354C5F9  mpp_mod_mp_mpp_er          72  mpp_util_mpi.inc
> ufs_model          000000000342C5D1  fms_io_utils_mod_         190  fms_io_utils.F90
> ufs_model          000000000347DAB5  netcdf_io_mod_mp_         368  netcdf_io.F90
> ```
> 
> The rest of the tests passed on Jet. Do you know what might be causing this issue while testing this PR on Jet?

@MichaelLueken, this is related to namelist changes to the `nstf_name` parameter and associated required sfc_data.nc fields.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, thank you for your comment! How can I resolve this issue? Without the change of `nstf_name` to [2,0,0,0,0], the results are not bit-to-bit reproducible.

**Comment by JeffBeck-NOAA** [comment]
@chan-hoo, why was the FV3_GFS_v15p2 SDF changed to run with nstf_name: [2, 0, 0, 0, 0] and FV3_GFS_v16 SDF changed from nstf_name: [2, 1, 0, 0, 0] to nstf_name: [2, 0, 0, 0, 0]?

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, With [2,1,0,0,0], the results of the 'restart' run are not identical with those of the original run. The restart reproducibility is the critical requirement of NCO.

**Comment by chan-hoo** [comment]
Only two ccpp suites `FV3_GFS_v16` and `FV3_GFS_v15p2` are available in Online-CMAQ (AQM v7).

**Comment by MichaelLueken** [comment]
The rest of the Jenkins tests have successfully completed for Cheyenne Intel, Cheyenne GNU, Gaea, and Orion.  Once the issue on Jet has been addressed, I will retest and this work can be merged.

**Comment by RatkoVasic-NOAA** [comment]
I don't know if we use this, or if this should be used at all?
```
    integer              :: nstf_name(5)    !< flag 0 for no nst  1 for uncoupled nst  and 2 for coupled NST
                                            !< nstf_name contains the NSST related parameters
                                            !< nstf_name(1) : 0 = NSSTM off, 1 = NSSTM on but uncoupled, 2 = NSSTM on and coupled
                                            !< nstf_name(2) : 1 = NSSTM spin up on, 0 = NSSTM spin up off
                                            !< nstf_name(3) : 1 = NSST analysis on, 0 = NSSTM analysis off
                                            !< nstf_name(4) : zsea1 in mm
                                            !< nstf_name(5) : zsea2 in mm
```

**Comment by JeffBeck-NOAA** [comment]
> I don't know if we use this, or if this should be used at all?
> 
> ```
>     integer              :: nstf_name(5)    !< flag 0 for no nst  1 for uncoupled nst  and 2 for coupled NST
>                                             !< nstf_name contains the NSST related parameters
>                                             !< nstf_name(1) : 0 = NSSTM off, 1 = NSSTM on but uncoupled, 2 = NSSTM on and coupled
>                                             !< nstf_name(2) : 1 = NSSTM spin up on, 0 = NSSTM spin up off
>                                             !< nstf_name(3) : 1 = NSST analysis on, 0 = NSSTM analysis off
>                                             !< nstf_name(4) : zsea1 in mm
>                                             !< nstf_name(5) : zsea2 in mm
> ```

@RatkoVasic-NOAA, @chan-hoo, I believe the problem is that when 2,0,0,0,0 is used, nstf_name(2) is set to 0, and therefore, NSSTM spin up is off (because it's a restart configuration).  In that case, tref is a required variable that is read in from the restart files.  The WE2E tests are not configured to run as restarts, so a single namelist is not going to work when we have both cold-start and restart configurations being used.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, it sounds like the NSSTM spin up has a restart reproducibility issue. Who can we contact to resolve this issue?

**Comment by RatkoVasic-NOAA** [comment]
>  The WE2E tests are not configured to run as restarts, so a single namelist is not going to work when we have both cold-start and restart configurations being used.
@JeffBeck-NOAA @chan-hoo , does it mean it can work with one value for cold start and another value for the restart?

**Comment by JeffBeck-NOAA** [comment]
> > @JeffBeck-NOAA @chan-hoo , does it mean it can work with one value for cold start and another value for the restart?

@RatkoVasic-NOAA, @chan-hoo, my guess is that we will need new restart WE2Es that use a different namelist, since we can't use a single namelist for both cold-start and restart tests.  One option would be to have the restart WE2E reset the nstf_name entry prior to running.  Another option would be to have a separate restart-specific YAML file to contain the necessary namelists for those tests.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, as @MichaelLueken said, if the failure was temporary and he runs the case successfully on Jet, it should not be an issue on `nstf_name`. Since the restart reproducibility is not guaranteed with [2,1,0,0,0], it would be more reasonable to have [2,0,0,0,0] by default until we don't have the restart reproducibility issue with [2,1,0,0,0] any more.

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA, as @MichaelLueken said, if the failure was temporary and he runs the case successfully on Jet, it should not be an issue on `nstf_name`. Since the restart reproducibility is not guaranteed with [2,1,0,0,0], it would be more reasonable to have [2,0,0,0,0] by default until we don't have the restart reproducibility issue with [2,1,0,0,0] any more.

@chan-hoo, unfortunately, changing "nstf_name" to 2,0,0,0,0 in the namelists will stop the App from working in cold-start mode (chgres_cube is not set up to include "tref", and I don't believe it's available in most external model data).  This is why I was saying earlier that we need separate restart and cold-start WE2Es using different namelists.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, I can see the same issue on Hera too. I'll add a new flag to the new script `update_restart_input_nml_file.py` for this case. Is this only in the case of ICS/LBCS="GSMGFS"? or are there any others calling "tref"?

**Comment by chan-hoo** [comment]
> This is why I was saying earlier that we need separate restart and cold-start WE2Es using different namelists.

@JeffBeck-NOAA, I think you are confusing the restart capability of this PR with the warm-start. The restart capability in this PR means that when a forecast run (`run_fcst`) failed at some point, the next try starts from the latest (nearest) restart files in the 'RESTART' directory. This is a function of the ufs weather model and the next run does use the same IC/LBC files.

**Comment by chan-hoo** [comment]
@MichaelLueken, I've added change of nstf_name for `GSMGFS` to `update_input_nml.py`. Please run the fundamental tests again.

**Comment by JeffBeck-NOAA** [comment]
> > This is why I was saying earlier that we need separate restart and cold-start WE2Es using different namelists.
> 
> @JeffBeck-NOAA, I think you are confusing the restart capability of this PR with the warm-start. The restart capability in this PR means that when a forecast run (`run_fcst`) failed at some point, the next try starts from the latest (nearest) restart files in the 'RESTART' directory. This is a function of the ufs weather model and the next run does use the same IC/LBC files.

@chan-hoo, thanks for creating the new update namelist script!  I think that will solve the problem.  To respond to your previous message about GSMGFS data, if nstf_name is set to 2,0,0,0,0 in the namelist for a cold-start run, regardless of what ICs/LBCs are used, "tref" will not be found in the sfc_data.nc file because chgres_cube does not create that field.  For warm-start/restart runs, this isn't a problem, since "tref" is included in the restart files, so setting nstf_name to 2,0,0,0,0 in that case will work.  So, the nstf_name changes were perfectly fine for restart/warm-start runs, they just wouldn't be compatible when running the model in cold-start mode.

**Comment by JeffBeck-NOAA** [comment]
@chan-hoo, since the fundamental way to configure the App is for cold-start simulations, my suggestion is to keep nstf_name = 2,1,0,0,0 in the FV3.input.yml file (the default) and change the update script to set it to 2,0,0,0,0 when running a restart-specific simulation.  Therefore, the update script would be specifically for restarts.  The RRFS and AQM models are instantiations of the SRW App, so they shouldn't be the primary configuration.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, The restart reproducibility is the critical requirement for all configurations. It doesn't matter if a run is cold-start or warm-start, even RRFS or AQM. It should be applied across the app. As I mentioned above, the restart reproducibility is not guaranteed with [2,1,0,0,0]. This means the NSST spin-up option has an issue to be resolved. It would be better not to put an incomplete configuration by default into the app. If `GMSGFS` causes an issue with the restart option, actually we should resolve the issue by updating `UFS_UTILS` (not by modifying the app itself). The reason why I made a change only for 'FV3_GFS_v16', 'FV3_GFS_v15p2', and 'FV3_GFS_v15_thompson_mynn_lam3km' in this PR is that I tested them within this PR and others are not required by AQM at this moment. However, when RRFS is incorporated into the app, this should be tested and updated with other ccpp suits supported in RRFS again. In my opinion, [2,0,0,0,0] should be set by default for all ccpp suits in the app.

**Comment by JeffBeck-NOAA** [comment]
@chan-hoo, the setting of nstf_name does matter when it comes to cold-starts, restarts, or warm-starts.  Setting it to 2,0,0,0,0 is for restarts only, while setting it to 2,1,0,0,0 is for cold starts (or a restart run if someone wishes to spin-up NSST instead of use the 'tref' field in the restart files).  The App has used 2,1,0,0,0 up until now.  I'm not saying we can't set nstf_name to 2,0,0,0,0 for restarts, I'm just saying that should be done in the update script when a restart/warm-start run is being configured.   We have the capability to do that now that you've included that script.

> In my opinion, [2,0,0,0,0] should be set by default for all ccpp suits in the app.

Setting 2,0,0,0,0 as the default for all CCPP suites means that the default configuration would be a restart or warm-start run, which isn't the case for the SRW App.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, now I get what your point is. I'll test it again.

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA, now I get what your point is. I'll test it again.

Thanks, @chan-hoo!  We just need to make sure the correct nstf_name setting is used for each configuration of the App.  I think your new script will allow that to happen.  It is very concerning to hear that the NSST spin-up is not b4b identical, and we should absolutely discuss this with others to see how to correct it.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, I have another question. Do you think 'FV3_GFS_v17_p8' should have [2,1,0,0,0] as well for cold-start?

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA, I have another question. Do you think 'FV3_GFS_v17_p8' should have [2,1,0,0,0] as well for cold-start?

@chan-hoo, yes, I think so.

**Review by JeffBeck-NOAA — APPROVED** [review]
Thank you, @chan-hoo!  Everything looks good.

**Comment by chan-hoo** [comment]
@JeffBeck-NOAA, it works well !!! Thank you for your suggestion and I am sorry again for my late understanding! I am running a couple of  WE2E tests now.

**Comment by JeffBeck-NOAA** [comment]
> @JeffBeck-NOAA, it works well !!! Thank you for your suggestion and I am sorry again for my late understanding! I am running a couple of WE2E tests now.

@chan-hoo, excellent!  Glad to hear things are running well and no problem!

**Comment by chan-hoo** [comment]
@MichaelLueken, the latest change passed three WE2E tests for FV3_GFS_v16, FV3_GFS_v17_p8, and FV3_GFS_v15p2 (w/ GSMGFS). Could you check the results of the Jenkins test?

**Comment by MichaelLueken** [comment]
@chan-hoo Thank you for working with @JeffBeck-NOAA with respect to nstf_name and NSST spin-up issues!  I have resubmitted the Jenkins tests for this PR.  There are still issues with the Jenkins tests following Friday's migration to a new site, so I will let you know if there are any new failures.

**Comment by MichaelLueken** [comment]
@chan-hoo The Jenkins tests have passed.  I'm in the middle of manually rerunning the tests on Jet now.  I'll update the PR once the Jet tests are complete.

**Comment by MichaelLueken** [comment]
@chan-hoo The manual WE2E fundamental tests on Jet have successfully passed.

I have also spoken with @christinaholtNOAA.  We are hoping to get her PR in tomorrow or early on Friday.  If there are issues with the Jenkins testing or manual tests on Jet, then I will be able to move forward with these changes.  Thank you!

**Comment by chan-hoo** [comment]
@MichaelLueken, can you merge this PR first? I'd like to create a production branch for AQM from this PR without @christinaholtNOAA's PR.

**Comment by MichaelLueken** [comment]
@chan-hoo Unfortunately, this PR will only be able to be merged before @christinaholtNOAA's PR if there are still issues with her PR.  If you don't want to include @christinaholtNOAA's PR in the production branch for `aqmv7`, then I would suggest creating the `production/aqmv7` branch from this branch, or waiting and see if there are still issues with the testing following her latest updates, at which time I can move forward with this work.

**Comment by chan-hoo** [comment]
@MichaelLueken, no problem. Will do.

**Comment by chan-hoo** [comment]
@MichaelLueken, you can merge this PR although I found some bugs on WCOSS2 and AQM configuration caused by PR #676 (Issues https://github.com/ufs-community/ufs-srweather-app/issues/709 and https://github.com/ufs-community/ufs-srweather-app/issues/710). I hope @christinaholtNOAA will resolve them soon. This PR itself doesn't show any issues on Hera.

**Comment by MichaelLueken** [comment]
@chan-hoo From @christinaholtNOAA's discussion in issue #709, it sounds like she has a fix for several issues.  I will hold off on merging this work until both of you have decided the best path to get the updates in (either through this PR or in a separate PR).  Having said that, I have rerun the Jenkins tests after you updated your branch to the latest develop and all tests successfully passed.  So, these changes are ready to go either now, or after merging Christina's changes in.

**Comment by chan-hoo** [comment]
@MichaelLueken, Can you merge this PR? There are so many issues to be resolved for AQM now, so I think it would be better to open a separate PR for them. I'll have to review and merge the PR #708 for the AQM implementation first.

**Pull request merged**
